includes/Discovery.php

```diff
@@ -262,22 +303,58 @@ public function print_twitter_metadata() {
 		if ( ! $enable_twitter_metadata ) {
 			return;
 		}
-		?>
-		<meta name="twitter:card" content="summary_large_image" />
-		<?php
+		$metadata = $this->get_twitter_metadata();
 
-		if ( ! has_post_thumbnail() ) {
-			return;
+		foreach ( $metadata as $name => $value ) {
+			printf( '<meta name="%s" content="%s" />', esc_attr( $name ), esc_attr( $value ) );
 		}
+	}
 
-		$poster = wp_get_attachment_image_url( (int) get_post_thumbnail_id(), Media::POSTER_PORTRAIT_IMAGE_SIZE );
+	/**
+	 * Get Twitter card metadata.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return array
+	 */
+	protected function get_twitter_metadata() {
+		$metadata = [
+			'twitter:card' => 'summary_large_image',
+		];
 
-		if ( ! $poster ) {
-			return;
+		/**
+		 * We're expecting a post object.
+		 *
+		 * @var WP_Post $post
+		 */
+		$post = get_queried_object();
+
+		if ( $post instanceof WP_Post ) {
+			if ( has_post_thumbnail( $post ) ) {
+				$poster = wp_get_attachment_image_url( (int) get_post_thumbnail_id( $post ), Media::POSTER_PORTRAIT_IMAGE_SIZE );
+
+				if ( $poster ) {
+					$metadata['twitter:image']     = esc_url( $poster );
+					$metadata['twitter:image:alt'] = the_title_attribute(
```

Comment:
Doesn't this cause double escaping?

Reply:
Is that a problem?

Reply:
It was used [here](https://github.com/google/web-stories-wp/blob/7a9fe66c83713c5877bcbc4fc9855d86f4bf150b/includes/Discovery.php#L217)

Reply:
It might cause problems.

If everything is passed through `esc_attr()`, we can safely use `get_the_title()` here instead.

Reply:
Fixed in https://github.com/google/web-stories-wp/pull/5919/commits/78329de9f2793bc25961b6326446d19a6c76e0fd